### PR TITLE
en: Update doc for aurora migration

### PR DESCRIPTION
**Pull request by ireneontheway**
### What is changed, added or deleted? (Required)

improve aurora migration guide

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-dm/pull/302 https://github.com/pingcap/docs-dm/pull/380
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts after cherry-picked

### CODE REVIEW

**Comment by TomShawn** [comment]
@ireneontheway It is not recommended to add internal Google Docs links in PR because only PingCAP members can access these links.

File: en/hardware-and-software-requirements.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ DM can be deployed and run on a 64-bit generic hardware server platform (Intel x
 >
 > - In the test environment, DM-master and DM-worker used for functional verification can be deployed on the same server.
 > - To prevent interference with the accuracy of the performance test results, it is **not recommended** to use low-performance storage and network hardware configurations.
-> - If you need to verify the function only, you can deploy a DM-master on a single machine. The number of DM-worker deployed depends on the number of upstream MySQL instances to be migrated.
+> - If you need to verify the function only, you can deploy a DM-master on a single machine. The number of DM-worker deployed is equivalent to the number of upstream MySQL instances at least. To ensure high availability, it is recommended to deploy more DM-workers.
```
Comment (TomShawn) [inline en/hardware-and-software-requirements.md:39]:
```suggestion
> - If you need to verify the function only, you can deploy a DM-master on a single machine. The number of DM-worker deployed must be greater than or equal to the number of upstream MySQL instances. To ensure high availability, it is recommended to deploy more DM-workers.
```

File: en/migrate-from-mysql-aurora.md [hunk 2]
```diff
@@ -57,19 +57,21 @@ To ensure a successful migration, you need to do prechecks before starting the m
 
 ### DM nodes deployment 
 
-As the hub of data migration, DM needs to connect to the upstream Aurora cluster and the downstream TiDB cluster. Therefore, you need to use the MySQL client to check whether the nodes in which DM is to be deployed can connect to the upstream and downstream. In addition, for software and hardware requirements for DM deployments, see [DM Cluster Software and Hardware Recommendations](hardware-and-software-requirements.md).
+As the hub of data migration, DM needs to connect to the upstream Aurora cluster and the downstream TiDB cluster. Therefore, you need to use the MySQL client to check whether the nodes in which DM is to be deployed can connect to the upstream and downstream. In addition, for software and hardware requirements ,and the node number for DM deployments, see [DM Cluster Software and Hardware Recommendations](hardware-and-software-requirements.md).
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:60]:
```suggestion
As the hub of data migration, DM needs to connect to the upstream Aurora cluster and the downstream TiDB cluster. Therefore, you need to use the MySQL client to check whether the nodes in which DM is to be deployed can connect to the upstream and downstream. In addition, for details of DM requirements on hardware, software, and the node number, see [DM Cluster Software and Hardware Recommendations](hardware-and-software-requirements.md).
```

File: en/migrate-from-mysql-aurora.md [hunk 3]
```diff
@@ -57,19 +57,21 @@ To ensure a successful migration, you need to do prechecks before starting the m
...
 
 ### Aurora
 
 DM relies on the `ROW`-formatted binlog for incremental replication. See [Enable binary for an Aurora Cluster](https://aws.amazon.com/premiumsupport/knowledge-center/enable-binary-logging-aurora/?nc1=h_ls) for the configuration instruction.
 
-To migrate data based on GTID, set both `gtid-mode` and `enforce_gtid_consistency` to `ON`. See [Configuring GTID-Based Replication for an Aurora MySQL Cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/mysql-replication-gtid.html#mysql-replication-gtid.configuring-aurora) for details.
+If Aurora enables GTID, then you can migrate data based on it. For the method to enable GTID, see [Configuring GTID-Based Replication for an Aurora MySQL Cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/mysql-replication-gtid.html#mysql-replication-gtid.configuring-aurora). To migrate data based on GTID, you need to set the `enable-gtid` in the data source configuration file of step 3 to `true`.
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:66]:
```suggestion
If GTID is enabled in Aurora, you can migrate data based on GTID. For how to enable it, see [Configuring GTID-Based Replication for an Aurora MySQL Cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/mysql-replication-gtid.html#mysql-replication-gtid.configuring-aurora). To migrate data based on GTID, you need to set `enable-gtid` to `true` in the configuration file of data source in [step 3](#step-3-configure-the-data-source).
```

File: en/migrate-from-mysql-aurora.md [hunk 4]
```diff
@@ -57,19 +57,21 @@ To ensure a successful migration, you need to do prechecks before starting the m
...
 
 > **Note:**
 >
 > GTID-based data migration requires MySQL 5.7 (Aurora 2.04) version or later.
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:70]:
```suggestion
> + GTID-based data migration requires MySQL 5.7 (Aurora 2.04) version or later.
> + In addition to the Aurora-specific configuration above, the upstream database must meet other requirements for migrating from MySQL, such as table schemas, character sets, and privileges. See [Checking Items](precheck.md#checking-items) for details.
```

File: en/migrate-from-mysql-aurora.md [hunk 5]
```diff
@@ -57,19 +57,21 @@ To ensure a successful migration, you need to do prechecks before starting the m
...
 
-In addition to the Aurora-specific configuration above, the upstream database must meet other requirements for migrating from MySQL. See [Checking Items](precheck.md#checking-items).
+> **Note:**
+>
+> In addition to the Aurora-specific configuration above, the upstream database must meet other requirements for migrating from MySQL, such as table schemas, character sets and privileges. See [Checking Items](precheck.md#checking-items).
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:74]:
```suggestion
```

File: en/migrate-from-mysql-aurora.md [hunk 6]
```diff
@@ -132,7 +134,10 @@ The content of `source1.yaml`:
 ```yaml
 # Aurora-1
 source-id: "aurora-replica-01"
+
+# To migrate data based on GTID, you need to set this items to true.
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:138]:
```suggestion
# To migrate data based on GTID, you need to set this item to true.
```

File: en/migrate-from-mysql-aurora.md [hunk 7]
```diff
@@ -225,10 +232,14 @@ mydumpers:
 
 Start the task using `dmctl` through TiUP.
 
+> **Note:**
+>
+> Currently, when using `dmctl` through TiUP, you need to use the absolute path of `task.yaml`. TiUP properly supports relative paths in subsequent updates.
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:237]:
```suggestion
> Currently, when using `dmctl` in TiUP, you need to use the absolute path of `task.yaml`. TiUP will support the relative path in later versions.
```

File: en/migrate-from-mysql-aurora.md [hunk 8]
```diff
@@ -254,6 +265,18 @@ If the task is successfully started, the following information is returned:
 }
 ```
 
+If `source db replication privilege checker` and `source db dump privilege checker` errors are in the returned information, check whether unrecognized privileges exits in the `errorMsg` field. For example: 
+
+```
+line 1 column 287 near \"INVOKE LAMBDA ON *.* TO...
+```
+
+The returned information above shows that the `INVOKE LAMBDA` privilege causes an error. If the privilege is unique to Aurora, add the following content into the configuration file to skip the check. DM improves the automatic processing of Aurora privileges in version updates.
```
Comment (TomShawn) [inline en/migrate-from-mysql-aurora.md:274]:
```suggestion
The returned information above shows that the `INVOKE LAMBDA` privilege causes an error. If the privilege is Aurora-specific, add the following content to the configuration file to skip the check. DM will improve the automatic handling of Aurora privileges in later versions.
```

File: en/migrate-from-mysql-aurora.md [hunk 9]
```diff
@@ -57,19 +57,19 @@ To ensure a successful migration, you need to do prechecks before starting the m
...
+If GTID is enabled in Aurora, you can migrate data based on GTID. For how to enable it, see [Configuring GTID-Based Replication for an Aurora MySQL Cluster](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/mysql-replication-gtid.html#mysql-replication-gtid.configuring-aurora). To migrate data based on GTID, you need to set `enable-gtid` to `true` in the configuration file of data source in [step 3](#step-3-configure-the-data-source).
 
 > **Note:**
 >
-> GTID-based data migration requires MySQL 5.7 (Aurora 2.04) version or later.
+> + GTID-based data migration requires MySQL 5.7 (Aurora 2.04) version or later.
+> + In addition to the Aurora-specific configuration above, the upstream database must meet other requirements for migrating from MySQL, such as table schemas, character sets, and privileges. See [Checking Items](precheck.md#checking-items) for details.
 
-In addition to the Aurora-specific configuration above, the upstream database must meet other requirements for migrating from MySQL. See [Checking Items](precheck.md#checking-items).
```
Comment (ireneontheway) [inline en/migrate-from-mysql-aurora.md:73]:
```suggestion
```

**Comment by ireneontheway** [comment]
@TomShawn PTAL

**Review by TomShawn — APPROVED** [review]
LGTM

File: en/migrate-from-mysql-aurora.md [hunk 10]
```diff
@@ -254,6 +262,18 @@ If the task is successfully started, the following information is returned:
 }
 ```
 
+If `source db replication privilege checker` and `source db dump privilege checker` errors are in the returned information, check whether unrecognized privileges exits in the `errorMsg` field. For example:
```
Comment (lance6716) [inline en/migrate-from-mysql-aurora.md:265]:
```suggestion
If `source db replication privilege checker` and `source db dump privilege checker` errors are in the returned information, check whether unrecognized privileges exsit in the `errorMsg` field. For example: 
```

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
LGTM

**Pull request merged**